force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,9 +128,30 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        List<Opportunity> newOpp = new List<Opportunity>();
+        List<Account> findAcc = [SELECT Id FROM Account WHERE Name= :accountName];
+        Account acc = new Account();
+        if(findAcc.isEmpty()) {
+            acc.Name = accountName;
+            insert acc;
+        } else acc = [SELECT Id FROM Account WHERE Name= :accountName];
+
+        List<Opportunity> opp = [SELECT Id FROM Opportunity WHERE Name IN :oppNames];
+        for (String s: oppNames) {
+            if(opp.isEmpty()) {
+                Opportunity createNewOpp = new Opportunity();
+                createNewOpp.Name = s;
+                createNewOpp.AccountId = acc.Id;
+                createNewOpp.StageName = 'Prospect';
+                createNewOpp.CloseDate = Date.today();
+                newOpp.add(createNewOpp);
+            } else {
+                Opportunity createNewOpp = new Opportunity();
+                createNewOpp = [SELECT Id FROM Opportunity WHERE Name = :s];
```

Comment:
This needs some work still. You are querying for the list of opportunities that have the past in opp names. Then you loop through each but check if the list if empty in order to create the new opportunity for each. What if some are found but some are not? Next in the else statement, you have a query inside a loop. You have already queries for the opportunities, why are you querying them again individually?